src/02_spec/07_modules/dem_uart/index.rst

```diff
@@ -0,0 +1,36 @@
+UART Device Emulation Module (DEM-UART)
+=======================================
+
+.. figure:: img/osd_dem_uart.*
+   :alt: Device Emulation Module UART
+
+The UART Device Emulation Module connects to the bus of a given CPU on one side,
+and to the Debug Interconnect on the other.
+Towards the CPU it behaves like a UART-16550A device, but instead of transmitting information
+over a UART-Interface it instead passes it to a host PC via the Debug Interconnect.
+
+The following UART registers are implemented and accessible via the bus, the address mapping
+is according to the `UART 16550 Datasheet <http://caro.su/msx/ocm_de1/16550.pdf>`_.
+The implemented properties are specified in the `OpenRISC Device Tree <https://elixir.bootlin.com/linux/latest/source/arch/openrisc/boot/dts/or1ksim.dts>`_,
+See `this <https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/serial/8250.txt>`_ for more information about the Device Tree specification.
+No Modem or DMA-Mode related features, registers or interrupts are implemented.
```

Comment:
You're mixing a couple of things here. In this documentation you specify what your module can and cannot do. Linux (and other software) is then a user of this module and the documentation you're writing.

So: please specify exactly how your UART module behaves. If it fully conforms to a spec (like the SIDSA one you linked), mention that. If it implements only some parts of this spec, mention that as well. If the referenced spec gives multiple options for some settings and you implement only one, specify that.

Your reference link (http://caro.su/msx/ocm_de1/16550.pdf) goes to a non-official link of a 16550 UART, not a 16550A one. Please discuss what exact features you support, and from which device version they are taken.

From the view of this spec, it's a coincidence that the parts we implement happen to match what OpenRISC Linux uses as default.